connectors/source.py

```diff
@@ -92,6 +124,14 @@ def get(self, key, default=None):
     def has_field(self, name):
         return name in self._config
 
+    def any_field_is_empty(self, names):
+        if any(not self.has_field(name) for name in names):
+            raise ConfigurationFieldMissingError
```

Comment:
Should this error mention which field is missing? Ideally it should include all fields that are missing IMO